source/fe/mapping_q_generic.cc

```diff
@@ -3621,82 +3623,70 @@ namespace
    * Boundary<dim,spacedim>.
    */
   template <int dim, int spacedim, class TriaIterator>
-  void get_intermediate_points_on_object(const Manifold<dim, spacedim> &manifold,
-                                         const QGaussLobatto<1>        &line_support_points,
-                                         const TriaIterator &iter,
-                                         std::vector<Point<spacedim> > &points)
+  void add_intermediate_points(const Manifold<dim, spacedim> &manifold,
+                               const Table<2,double>         &weight_table,
+                               const TriaIterator            &iter,
+                               std::vector<Point<spacedim> > &points)
   {
     const unsigned int structdim = TriaIterator::AccessorType::structure_dimension;
 
     // Try backward compatibility option.
-    if (const Boundary<dim,spacedim> *boundary
-        = dynamic_cast<const Boundary<dim,spacedim> *>(&manifold))
+    const Boundary<dim,spacedim> *boundary
+      = dynamic_cast<const Boundary<dim,spacedim> *>(&manifold);
+    if (structdim < spacedim && boundary != NULL)
       // This is actually a boundary. Call old methods.
       {
+        std::vector<Point<spacedim> > new_points(weight_table.size(0));
         switch (structdim)
           {
           case 1:
           {
             const typename Triangulation<dim,spacedim>::line_iterator line = iter;
-            boundary->get_intermediate_points_on_line(line, points);
-            return;
+            boundary->get_intermediate_points_on_line(line, new_points);
+            break;
           }
           case 2:
           {
             const typename Triangulation<dim,spacedim>::quad_iterator quad = iter;
-            boundary->get_intermediate_points_on_quad(quad, points);
-            return;
+            boundary->get_intermediate_points_on_quad(quad, new_points);
+            break;
           }
           default:
             Assert(false, ExcInternalError());
-            return;
+            break;
           }
+        for (unsigned int i=0; i<new_points.size(); ++i)
+          points.push_back(new_points[i]);
```

Comment:
Not important, but you can do this more efficiently by saying
`points.insert (new_points.begin(), new_points.end(), points.end());`